src/controllers/addon_config.rs

```diff
@@ -454,6 +461,46 @@ impl FleetAddonConfig {
 
         Ok(None)
     }
+
+    async fn update_config_map(&self, ctx: Arc<Context>) -> ConfigMapSyncResult<()> {
+        let feature_gates = self
+            .spec
+            .config
+            .as_ref()
+            .map(|c| c.feature_gates.clone().unwrap_or_default())
+            .unwrap_or_default();
+
+        // If a ConfigMap reference is defined
+        if let Some(reference) = feature_gates.config_map.as_ref()
```

Comment:
For the use-case we are trying to address, we can expect the `ConfigMap` to be always present, so we can use this here:
```rust
let mut config_map: ConfigMap = client.fetch(reference).await?;
```